### PR TITLE
Allow users to add songs to their on queue playlists 

### PR DESCRIPTION
**Pull request by luisa-li**
Changed the user creation workflow to create a default on queue playlist for them. 
Added endpoint for user to add a track to their on queue playlist. 

## Description
See above.
[Link to Ticket](https://github.com/GenerateNU/platnm/issues/104)

## How Has This Been Tested?
Postman screenshots, see below!

## Checklist:
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

## Screenshots:

Successful call to the create user endpoint. 
Screenshot 2024-10-24 at 11 12 42 PM
Corresponding created row for the on queue playlist in the playlist table.
Screenshot 2024-10-24 at 11 15 47 PM
Sending a post request to the end point that adds to the on queue playlist:
Screenshot 2024-10-24 at 11 30 52 PM
The resulting playlist track table: 
Screenshot 2024-10-24 at 11 30 18 PM

### CODE REVIEW

File: backend/internal/service/server.go [hunk 1]
```diff
@@ -82,6 +83,11 @@ func setupRoutes(app *fiber.App, config config.Config) {
 		r.Patch("/:recommendationId", recommendationHandler.ReactToRecommendation)
 	})
 
+	playlistHandler := playlist.NewHandler(repository.Playlist)
+	app.Route("/playlist", func(r fiber.Router) {
+		r.Post("/on_queue/:id", playlistHandler.AddToUserOnQueue)
```
Comment (ddusichka) [inline backend/internal/service/server.go:88]: this is subtle but can we call the parameter `:userId` so it's clear that it's not the trackId? (make sure to update it in the handler too)

File: backend/internal/storage/postgres/schema/playlist.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	return nil
+
+}
+
+func (r *PlaylistRepository) AddToUserOnQueue(ctx context.Context, id string, track models.Track) error {
+
+	var playlistID string
+
+	findQuery := `SELECT id FROM playlist WHERE user_id = $1 AND title = 'On Queue'`
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/playlist.go:34]: this made me think of a tiny edge case where a user could potentially create their own playlist also called "On Queue" and mess this up

(no action items for you but we should discuss as a team if we wanna ban this as a custom playlist name or handle it some other way in the future)

File: backend/internal/storage/postgres/schema/playlist.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	err := r.QueryRow(ctx, findQuery, id).Scan(&playlistID)
+
+	if err != nil {
+		return err
+	}
+
+	// now insert the track into the user's on queue playlist
+	insertQuery := `
+	INSERT INTO playlist_track (playlist_id, track_id)
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/playlist.go:43]: what happens if the track id is not a real track we have? 👀
i fear you might have a foreign key error, you can look at the CreateComment method for inspo on handling it

**Review by ddusichka — COMMENTED** [review]
looks really good!! just two quick suggestions

File: backend/internal/storage/postgres/schema/playlist.go (hunk 2 above)
Comment (luisa-li) [reply backend/internal/storage/postgres/schema/playlist.go:34]: makes sense!

File: backend/internal/service/server.go (hunk 1 above)
Comment (luisa-li) [reply backend/internal/service/server.go:88]: Changed this!

File: backend/internal/storage/postgres/schema/playlist.go (hunk 3 above)
Comment (luisa-li) [reply backend/internal/storage/postgres/schema/playlist.go:43]: fixed this also! here is the screenshot of a postman call with a not real id!
Screenshot 2024-10-25 at 11 59 18 PM

**Review by ddusichka — APPROVED** [review]
great job!

**Pull request merged**